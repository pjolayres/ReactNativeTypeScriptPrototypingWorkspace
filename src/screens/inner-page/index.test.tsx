import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { InnerPage } from '.';

// jest.mock('react-native-vector-icons/FontAwesome', () => 'Icon');

beforeAll(async () => {
  jest.setTimeout(30 * 1000); // 30-second timeout
});

it('renders correctly', () => {
  const title = 'title';
  const navigation: any = {
    getParam: jest.fn(() => title)
  };
  const setName = jest.fn<(name: string) => void>();
  const setVersion = jest.fn<(version: string) => void>();

  const tree = renderer.create(<InnerPage setName={setName} setVersion={setVersion} navigation={navigation} name="Test Name" version="Test Version" />).toJSON();

  expect(tree).toMatchSnapshot();
});
