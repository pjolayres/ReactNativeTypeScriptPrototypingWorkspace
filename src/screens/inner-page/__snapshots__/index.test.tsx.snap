// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    title
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    --------
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      Go Back
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    --------
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Name: 
    Test Name
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Version: 
    Test Version
  </Text>
</View>
`;
